stdlib/messaging/rabbitmq/src/main/java/org/ballerinalang/messaging/rabbitmq/util/ConnectionUtils.java

```diff
@@ -97,18 +96,68 @@ public static Connection createConnection(BMap<String, BValue> connectionConfig)
+     *                     operations, use -1 for infinity.
+     * @param closeCode    The close code (See under "Reply Codes" in the AMQP specification).
+     * @param closeMessage A message indicating the reason for closing the connection.
+     * @param context      Context.
      */
-    public static void handleCloseConnection(Connection connection, BValue timeout, Context context) {
+    public static void handleCloseConnection(Connection connection, BValue closeCode, BValue closeMessage,
+                                             BValue timeout, Context context) {
         boolean validTimeout = timeout instanceof BInteger;
-        if (!validTimeout) {
-            closeConnection(connection, context);
+        boolean validCloseCode = (closeCode instanceof BInteger) && (closeMessage instanceof BString);
+        if (!validTimeout && !validCloseCode) {
+            close(connection, context);
+        } else if (validTimeout && validCloseCode) {
+            close(connection,
+                    Math.toIntExact(((BInteger) closeCode).intValue()),
+                    closeMessage.stringValue(),
+                    Math.toIntExact(((BInteger) timeout).intValue()),
+                    context);
+        } else if (validTimeout && !validCloseCode) {
+            close(connection,
+                    Math.toIntExact(((BInteger) timeout).intValue()),
+                    context);
+        } else if (!validTimeout && validCloseCode) {
+            close(connection,
+                    Math.toIntExact(((BInteger) closeCode).intValue()),
+                    closeMessage.stringValue(),
+                    context);
         } else {
-            closeConnection(connection,
-                    Math.toIntExact(((BInteger) timeout).intValue()), context);
+            throw new BallerinaException("Error in closing the connection");
+        }
+    }
+
+    /**
+     * Handles aborting the given connection.
+     *
+     * @param connection   RabbitMQ Connection object.
+     * @param timeout      Timeout (in milliseconds) for completing all the close-related
+     *                     operations, use -1 for infinity.
+     * @param closeCode    The close code (See under "Reply Codes" in the AMQP specification).
+     * @param closeMessage A message indicating the reason for closing the connection.
+     */
+    public static void handleAbortConnection(Connection connection, BValue closeCode, BValue closeMessage,
+                                             BValue timeout) {
+        boolean validTimeout = timeout instanceof BInteger;
+        boolean validCloseCode = (closeCode instanceof BInteger) && (closeMessage instanceof BString);
+        if (!validTimeout && !validCloseCode) {
+            abortConnection(connection);
+        } else if (validTimeout && validCloseCode) {
+            abortConnection(connection,
+                    Math.toIntExact(((BInteger) closeCode).intValue()),
```

Comment:
When this method is called it may throw an `ArithmeticException`, shouldn't we be handling that as well?

Reply:
Resolved.